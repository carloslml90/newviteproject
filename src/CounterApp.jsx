import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp=({value}) => {
console. log ('executing')
    const [counter, SetCounter]=useState(value);

    const CounterAdd=() => { 

        SetCounter(

        (c) => c + 1);
    }

    const CounterSubstract=() => {

        SetCounter(

            (c) => counter -1
        );
    }    

    const CounterReset=() => {

        SetCounter(

            (c) => value
        );
    }    




    return (
    <>
        <h1>Counter APP</h1>

       

        <h2> Cups of coffe today {counter}</h2>
        <button onClick={CounterAdd}> +1 </button>
        <button onClick={CounterSubstract}> -1 </button>
        <button onClick={CounterReset}> Reset </button>

    </> 
   
        )
}

CounterApp.propTypes={

value: PropTypes.number.isRequired,
}

