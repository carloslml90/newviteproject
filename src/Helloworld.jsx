

import PropTypes from 'prop-types';
export const Hello =({title, description, goal}) => {
 const  NewFunction=() =>{
console.log('hola mundo');
}
return (
<>
    <h1 data-testid="test-title">{title}</h1>
    <p>{goal}</p>
    <p>{description}</p>


    </>
)



}


Hello.propTypes= {
title: PropTypes.string,
description: PropTypes.string,
goal: PropTypes.string





}