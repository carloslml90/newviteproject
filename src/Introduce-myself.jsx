


const name= 'carlos';

const apellido='mata'

const fullname= name.toUpperCase()+' '+ apellido.toUpperCase();

const dates={age:35, stature: 5.7 }


export const MyIntroduce=() =>{


    return ( 

<p> Hello my name is {fullname} and my data are the following: <code>{JSON.stringify(dates)}</code> </p>

        )
    



}