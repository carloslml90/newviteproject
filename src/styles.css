p {font-size: 5 em ; font-family: Verdana, Geneva, Tahoma, sans-serif; font-style: oblique; color: rosybrown };