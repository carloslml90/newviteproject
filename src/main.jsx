import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App'
import { Hello } from './Helloworld'
import { MyIntroduce } from './Introduce-myself'
import { CounterApp } from './CounterApp'


import "./styles.css"
/* import './index.css' */



ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={50} />

 <Hello 
 title='i love me a lot'
 description='nothing interesting'
 goal='Will be a great progammer'/>
 
  </React.StrictMode>,
)
