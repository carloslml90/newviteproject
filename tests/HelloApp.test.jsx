import { render, screen } from "@testing-library/react";
import { Hello } from "../src/Helloworld";


describe('<FirstApp/> tests',()=>{


    /* test('should do match with snapshot',()=>{
        
        const nombre='carlos';    
        const title=`${nombre} this is the best web developer  in the world`;
        const {container}=render(<Hello title={title}/>);
        console.log(container);

        expect(container).toMatchSnapshot();
        
        
    }) */
    
    const title='The best developer';

    test('should show title inside a h1',()=>{

    
    const {container, getByText}=render(<Hello title={title}/>)

    expect(getByText).toBeTruthy();
    expect(container).toMatchSnapshot();
    const h1=container.querySelector('h1');
    expect(h1.innerHTML).toContain(title);
})

/* test('I want to create another snapshot', ()=>{

const title='i am practicing for being a big developer';
const description= 'embrace the struggle and just do it being strong and brave '
const {container}= render(<Hello title={title} description={description}/>)

expect(container).toMatchSnapshot();
}) */

test('it must show a message saying "i love me a lot ',()=>{

render(<Hello title={title}/>)
expect(screen.getByText(title)).toBeTruthy();
})
})