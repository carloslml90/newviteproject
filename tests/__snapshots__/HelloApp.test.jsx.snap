// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstApp/> tests should show title inside a h1 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    The best developer
  </h1>
  <p />
  <p />
</div>
`;
